steel/src/contract.rs

```diff
@@ -286,13 +289,15 @@ where
     ///
     /// [EvmEnv]: crate::EvmEnv
     pub fn call(self) -> C::Return {
+        // safe unwrap: env is never returned without a DB
         let state_db = self.env.db.as_ref().unwrap();
         let mut evm = new_evm::<_, H>(
             WrapStateDb::new(state_db),
             self.env.cfg_env.clone(),
             self.env.header.inner(),
         );
-        self.tx.transact(&mut evm).unwrap()
+        // typically, guest functions should panic instead of returning an error
+        self.tx.transact(&mut evm).expect("call failed")
```

Comment:
I wouldn't say this is true. We do use panic liberally in guest code, and in the `main` especially, but guest libraries can use errors instead, and generally should, I think, as that they can optionally be handled by the caller.

I think in this case, I would return the result produced by `self.tx.transact`